src/Mjolnir.ts

```diff
@@ -196,14 +166,15 @@ export class Mjolnir {
         }
 
         const ruleServer = config.web.ruleServer ? new RuleServer() : null;
-        const mjolnir = new Mjolnir(client, managementRoomId, config, protectedRooms, policyLists, ruleServer);
-        await mjolnir.logMessage(LogLevel.INFO, "index", "Mjolnir is starting up. Use !mjolnir to query status.");
+        const mjolnir = new Mjolnir(client, await client.getUserId(), managementRoomId, config, protectedRooms, policyLists, ruleServer);
+        await mjolnir.managementRoom.logMessage(LogLevel.INFO, "index", "Mjolnir is starting up. Use !mjolnir to query status.");
         Mjolnir.addJoinOnInviteListener(mjolnir, client, config);
         return mjolnir;
     }
 
     constructor(
         public readonly client: MatrixClient,
+        private readonly clientUserId: string,
```

Comment:
The problem is that we can't call that method in a constructor and we would otherwise need to or find a way around it.

Reply:
> The problem is that we can't call that method in a constructor and we would otherwise need to or find a way around it.

I understand. But now that we have `clientUserId`, we can probably remove many calls to `await mjolnir.client.getUserId()`, no?

Reply:
I don't really want to, I'm still not sure it's the right thing.